protocol/src/packet.rs

```diff
@@ -0,0 +1,87 @@
+use std::io::{Cursor, Read, Write};
+
+use crate::decoder::{Decoder, DecoderReadExt};
+use crate::encoder::{Encoder, EncoderWriteExt};
+use crate::error::{DecodeError, EncodeError};
+
+#[derive(Debug)]
+pub struct Packet {
```

Comment:
When you will intergrate library in a network library like tokio this structure would not be necessary. There are more steps in packet codec. Like encryption and compression.

Reply:
> When you will intergrate protocol in a network library like tokio this structure would not be necessary.

Do you have an example for how this works?

I am currently converting a `tokio::net::TcpStream` into `std::net::TcpStream` so I can use `Packet::decode` with it, so this would definitely be an improvement over what I have now.